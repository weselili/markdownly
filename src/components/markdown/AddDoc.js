import React, { useState } from 'react';
import PropTypes from 'prop-types';

function AddDoc({ handleSubmit }) {
  const [title, updateTitle] = useState('');

  return (
    <form onSubmit={(event) => handleSubmit(event, title)}>
      <input onChange={({ target }) => updateTitle(target.value)} type="text" value={title} name="title" placeholder="title"></input>
      <button>Add Document</button>
    </form>
  );
}

AddDoc.propTypes = {
  handleSubmit: PropTypes.func.isRequired
};

export default AddDoc;
