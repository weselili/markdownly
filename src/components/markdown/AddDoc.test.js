import React from 'react';
import { shallow } from 'enzyme';
import AddDoc from './AddDoc';

describe('AddDoc component', () => {
  it('renders AddDoc', () => {
    const wrapper = shallow(<AddDoc title={''} handleSubmit={() => ({})} />);
    expect(wrapper).toMatchSnapshot();
  });
});
  
