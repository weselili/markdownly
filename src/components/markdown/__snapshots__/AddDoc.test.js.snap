// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddDoc component renders AddDoc 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="title"
    onChange={[Function]}
    placeholder="title"
    type="text"
    value=""
  />
  <button>
    Add Document
  </button>
</form>
`;
